packages/tools/devtools/devtools-core/src/data-visualization/SharedTreeVisualizer.ts

```diff
@@ -148,64 +155,109 @@ export function toVisualTree(tree: VisualSharedTreeNode): VisualChildNode {
 }
 
 /**
- * Concatenrate allowed types for `ObjectNodeStoredSchema` and `MapNodeStoredSchema`.
+ * Concatenate allowed types for `ObjectNodeStoredSchema` and `MapNodeStoredSchema`.
  */
-function concatenateTypes(fieldTypes: ReadonlySet<string>): string {
+export function concatenateTypes(fieldTypes: ReadonlySet<string>): string {
 	return [...fieldTypes].join(" | ");
 }
 
 /**
- * Returns the allowed fields & types for the object fields (e.g., `foo : string | number, bar: boolean`)
+ * Properties that describe schema constraints for a field in the tree
  */
-function getObjectAllowedTypes(schema: SimpleObjectNodeSchema): string {
-	const result: string[] = [];
+interface FieldSchemaProperties {
+	/**
+	 * Set of type names that are valid for this specific node position in the tree.
+	 * This is a subset of the types defined in treeDefinitions.
+	 */
+	allowedTypes: ReadonlySet<string> | undefined;
 
-	for (const [fieldKey, treeFieldSimpleSchema] of Object.entries(schema.fields)) {
-		const fieldTypes = treeFieldSimpleSchema.allowedTypes;
-		result.push(`${fieldKey} : ${concatenateTypes(fieldTypes)}`);
-	}
-
-	return `{ ${result.join(", ")} }`;
+	/**
+	 * Whether this field is required in its parent object schema.
+	 * Only meaningful for direct children of object nodes.
+	 * Undefined for array/map elements since they are always required within their parent.
+	 */
+	isRequired: boolean | undefined;
 }
 
 /**
- * Returns the schema & fields of the node.
+ * Processes and visualizes the fields of a verbose tree node.
```

Comment:
Curious: what's a "verbose" tree node (vs a "not verbose" one)?

Reply:
It's a particular representation you can generate from a SharedTree that is:
- serializable
- contains sufficient schema information to reconstitute a SharedTree from it

See https://github.com/microsoft/FluidFramework/blob/9562c57f5e1eb062188b058ca916dea29834ea1a/packages/dds/tree/src/simple-tree/api/verboseTree.ts.

That said, I don't know that we need to call it out specifically in the docs here, Should be fine to just say something like "Visualizes a tree node".